cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/MockCruiseControl.java

```diff
@@ -124,66 +127,214 @@ public boolean isRunning() {
-                                .withSecure(true))
-                .respond(
-                        response()
-                                .withBody(json)
-                                .withHeaders(header("User-Task-ID", "cruise-control-state"))
-                                .withDelay(TimeUnit.SECONDS, 0));
+        String userTaskId = verbose
+                ? USER_TASK_REBALANCE_NO_GOALS_VERBOSE_RESPONSE_UTID
+                : USER_TASK_REBALANCE_NO_GOALS_RESPONSE_UTID;
 
-        // Verbose response
-        JsonBody jsonVerbose = new JsonBody(ReadWriteUtils.readFileFromResources(getClass(), "/" + CC_JSON_ROOT + "CC-State-verbose.json"));
+        JsonBody jsonBody = new JsonBody(ReadWriteUtils.readFileFromResources(
+                getClass(), "/" + CC_JSON_ROOT + fileName));
 
-        server
-                .when(
-                        request()
-                                .withMethod("GET")
-                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.JSON.toString(), "true"))
-                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.VERBOSE.toString(), "true"))
-                                .withPath(CruiseControlEndpoints.STATE.toString())
-                                .withHeader(AUTH_HEADER)
-                                .withSecure(true))
-                .respond(
-                        response()
-                                .withBody(jsonVerbose)
-                                .withHeaders(header("User-Task-ID", "cruise-control-state-verbose"))
-                                .withDelay(TimeUnit.SECONDS, 0));
+        return response()
+                .withBody(jsonBody)
+                .withStatusCode(200)
+                .withHeaders(header("User-Task-ID", userTaskId))
+                .withDelay(TimeUnit.SECONDS, 0);
+    }
 
+    /**
+     * Sets up mocked response of the Cruise Control "User Tasks" endpoint.
+     *
+     * <p>
+     * This method configures the MockServer to return specific responses based on the provided task status.
+     *
+     * @param taskStatus The current {@link CruiseControlUserTaskStatus} of the simulated Cruise Control task.
+     *                   This determines whether the state response should reflect an active,
+     *                   an ongoing execution, or a completed task.
+     */
+    public void mockUserTasksEndpoint(CruiseControlUserTaskStatus taskStatus) {
+        List<Header> authHeader = List.of(AUTH_HEADER);
+        for (boolean verbose : List.of(false, true)) {
+            HttpRequest request = requestMatcher(authHeader, CruiseControlEndpoints.USER_TASKS, "true", String.valueOf(verbose));
+
+            switch (taskStatus) {
+                case ACTIVE:
+                    server
+                            .when(request)
+                            .respond(userTaskResponse("Active", verbose));
```

Comment:
You know what I am gonna say right? Why are using the raw string when you have the `CruiseControlUserTaskStatus` enum to handle that for you?
```suggestion
                            .respond(userTaskResponse(CruiseControlUserTaskStatus.ACTIVE.toString(), verbose));
```

Reply:
Fixed.